src/SearchableEntity.php

```diff
@@ -58,20 +65,47 @@ private function setId()
         }
 
         if (1 == count($ids)) {
-            $this->id = reset($ids);
-        } else {
-            $objectID = '';
-            foreach ($ids as $key => $value) {
-                $objectID .= $key . '-' . $value . '__';
-            }
-
-            $this->id = rtrim($objectID, '_');
+            $id = \reset($ids);
+            $this->id = \is_scalar($id) ? (string)$id : $this->serializeValue($id);
+
+            return;
+        }
+
+        $objectID = '';
+        foreach ($ids as $key => $value) {
+            $objectID .= $key . '-' . $value . '__';
         }
 
+        $this->id = rtrim($objectID, '_');
     }
 
-    public function getId()
+    /**
+     * @param object $value
+     *
+     * @return array|string
+     */
+    private function serializeValue($value, $context = [])
     {
-        return $this->id;
+        if ($this->useSerializerGroups) {
+            $context['groups'] = [Searchable::NORMALIZATION_GROUP];
+        }
+
+        if ($this->normalizer instanceof NormalizerInterface) {
+            return $this->normalizer->normalize($value, Searchable::NORMALIZATION_FORMAT, $context);
+        }
+
+        if ($this->normalizer instanceof ArrayTransformerInterface) {
+            return $this->normalizer->toArray($value);
+        }
+
+        throw new \LogicException(
```

Comment:
This may be a breaking change. Because the `$value` may implement `__string` method. Any ideas?

Reply:
The code today would not return anything if the the value couldn't be serialized. I guess throwing an exception could be considered an breaking change, but I think it's worth it. Silently failing is the worse 👍 
About `__toString`, it won't work because this method must return an array.

Reply:
@julienbourdeau Actually @nunomaduro is right here. We may check fo `__toString()` here or above where we call `is_scalar()`, anyway we need to support this ;)

The exception itself is well.... BC break if you assume silent-failure is a feature. For me that was actually a bug :)